src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java

```diff
@@ -333,4 +337,18 @@ public boolean isUserAllowedToGetResult(Exercise exercise, StudentParticipation
                 && (exercise.getAssessmentDueDate() == null || exercise.getAssessmentDueDate().isBefore(ZonedDateTime.now())) && result.getAssessor() != null
                 && result.getCompletionDate() != null;
     }
+
+    /**
+     * Checks if the user is only student in the course.
+     *
+     * If the exam is over and the results are not published, he is not allowed to request a result.
+     *
+     * @param exercise - Exercise that the result is requested for
+     * @param user - User that requests the result
+     * @return true if user is allowed to see the result, false otherwise
+     */
+    public boolean isAllowedToGetExamResult(Exercise exercise, User user) {
+        return !this.isOnlyStudentInCourse(exercise.getCourseViaExerciseGroupOrCourseMember(), user) || (!exercise.hasExerciseGroup()
+                || !exercise.getExerciseGroup().getExam().getEndDate().isBefore(ZonedDateTime.now()) || exercise.getExerciseGroup().getExam().isPublished());
```

Comment:
Maybe I'm too dumb for that, but I think this is really hard to understand. Maybe the following is easier to understand? Please double check as I'm not that confident that I really get what you check here.
```suggestion
        return this.isAtLeastTeachingAssistantInCourse(exercise.getCourseViaExerciseGroupOrCourseMember(), user) || (exercise.hasExerciseGroup()
                && exercise.getExerciseGroup().getExam().getEndDate().isBefore(ZonedDateTime.now()) && exercise.getExerciseGroup().getExam().isPublished());
```

Reply:
After committing your proposed changes I realised, that there was an error. Nevertheless I've adjusted this condition to make it easier to read and added a clause for each condition in the description